### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by sharifsalah**
Updated the readme file for the todo-mvp-contentproviders branch with the following changes:

* Consistent terminology and style.
* Added a new 'What you need' section to list relevant pre-requisite topics.
* Extensive links to other readme files in this project, as well as external reference documentation.
* New heading and structure.
* Updated formatting.
* Fixed various grammatical issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,122 +70,28 @@ In [TasksPresenter.java](https://github.com/googlesamples/android-architecture/b
...
-#### Ease of amending or adding a feature
 
-Similar to MVP Loaders
+## Comparing this sample
 
-#### Learning cost
+The following summary reviews how this solution compares to the todo-mvp base sample:
 
-Medium as the Loaders and Content Providers framework are not trivial.
+ * <b>Use of architectural frameworks, libraries, or tools: </b>None required.
```
Comment (JoseAlcerreca) [inline README.md]: Both loaders and content providers are tools

File: README.md [hunk 2]
```diff
@@ -1,31 +1,45 @@
...
+ * Add support for custom searches in your application.
+ * Develop widgets to access data in your application.
+ * Provide support for copying and pasting data between your application and other applications.
 
-  * Manage access to a structured set of data
-  * Content providers are the standard interface that connects data in one process with code running in another process
+Content providers can also be used to manage access to other types of storage including non-relational databases, or data stored on a file system. You can also use the [`FileProvider`](https://developer.android.com/reference/android/support/v4/content/FileProvider.html) subclass of [`ContentProvider`](https://developer.android.com/reference/android/content/ContentProvider.html) to facilitate secure sharing of files associated with an app. A [`FileProvider`](https://developer.android.com/reference/android/support/v4/content/FileProvider.html) can be used to create a `content://` URI for a file instead of a `file:///` URI. As with other versions of the [todo-mvp](https://github.com/googlesamples/android-architecture/tree/todo-mvp) app, this sample uses a SQLite database to provide data storage.
```
Comment (JoseAlcerreca) [inline README.md]: At least, it's too much for the introduction. Originally it was at the end of the document, where it makes more sense.

File: README.md [hunk 3]
```diff
@@ -54,122 +70,28 @@ In [TasksPresenter.java](https://github.com/googlesamples/android-architecture/b
...
...epository`](https://github.com/googlesamples/android-architecture/blob/todo-mvp-contentproviders/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepository.java) class does not return data to the presenter, but instead stores the tasks in the [`LocalDataSource`](https://github.com/googlesamples/android-architecture/blob/todo-mvp-contentproviders/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.java). A [`CursorLoader`](https://developer.android.com/reference/android/content/CursorLoader.html) automatically registers a [`ContentObserver`](https://developer.android.com/reference/android/database/ContentObserver.html) to reload data when changes are detected. A [`ContentProvider`](https://developer.android.com/reference/android/content/ContentProvider.html) can then notify observers of changes to the URI queried as illustrated in the following delete method from [`TasksProvider`](https://github.com/googlesamples/android-architecture/blob/todo-mvp-contentproviders/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksProvider.java).
```
Comment (JoseAlcerreca) [inline README.md]: "following delete method" missing

File: README.md [hunk 4]
```diff
@@ -54,122 +70,28 @@ In [TasksPresenter.java](https://github.com/googlesamples/android-architecture/b
...
 
-#### Conceptual complexity
+The use of the loaders and content providers introduces additional complexity in writing unit tests for the new functionality. This sample relies on the use of tests which need to be run on either a device or emulator.
```
Comment (JoseAlcerreca) [inline README.md]: Shouldn't this paragraph be in a new section?

File: README.md [hunk 5]
```diff
@@ -54,122 +70,28 @@ In [TasksPresenter.java](https://github.com/googlesamples/android-architecture/b
...
-SUM:                            82           1197           1797           4136
--------------------------------------------------------------------------------
-
-```
+Note that some aspects of the code in this version are simpler as loaders take care of asynchronous requests. The following table summarizes the amount of code required for this version of the application and provides a comparison to the todo-mvp base sample.
 
-### Maintainability
+| Language      | Number of files | Blank lines | Comment lines | Lines of code |
+| ------------- | --------------- | ----------- | ------------- | ------------- |
+| **Java**      |               48|         1100|           1460|           3535 (3450 in todo-mvP)|
```
Comment (JoseAlcerreca) [inline README.md]: s/mvP/mvp

File: README.md [hunk 6]
```diff
@@ -1,31 +1,45 @@
...
 
-<img src="https://github.com/googlesamples/android-architecture/wiki/images/mvp-contentproviders.png" alt="Diagram"/>
+You could then use content providers to support additional features that are not covered by this sample, providing the following possible benefits:
 
-The advantages of Content Providers, from the [Content Provider documentation page](http://developer.android.com/guide/topics/providers/content-providers.html), are:
+ * Allow you to securely share data stored in your application with other applications.
+ * Add support for custom searches in your application.
+ * Develop widgets to access data in your application.
```
Comment (sharifsalah) [inline README.md]: I don't see the distinction between 'to' and 'that' in this case but I've updated to read:

Develop widgets which access data in your app.

File: README.md [hunk 7]
```diff
@@ -1,31 +1,45 @@
...
+ * Provide support for copying and pasting data between your application and other applications.
```
Comment (sharifsalah) [inline README.md]: Removed

File: README.md (hunk 2 above)
Comment (sharifsalah) [inline README.md]: This addresses a concern you and I have both expressed that content providers are often misunderstood to be directly related to SQL storage. We had agreed that the FileProvider example illustrated that other types of storage are supported so I'm leaving this as is.

File: README.md (hunk 5 above)
Comment (sharifsalah) [inline README.md]: fixed

File: README.md (hunk 1 above)
Comment (sharifsalah) [inline README.md]: This is based on the original:

Use of architectural frameworks/libraries/tools:

No external frameworks.

I've updated it to read 'Loaders and content providers.'

File: README.md (hunk 4 above)
Comment (sharifsalah) [inline README.md]: No, creating a new section for two sentences is excessive and inconsistent with the other revised readme files.

File: README.md (hunk 3 above)
Comment (sharifsalah) [inline README.md]: Thanks and fixed.

**Pull request merged**